### PR TITLE
consolidate bootstrap docs

### LINKED ISSUE #90686: Consolidate bootstrap documentation
Right now, there are docs for x.py in a bunch of different places:
- https://rustc-dev-guide.rust-lang.org/building/how-to-build-and-run.html#building-the-compiler
- https://rustc-dev-guide.rust-lang.org/building/bootstrapping.html
- https://github.com/rust-lang/rust/blob/master/src/bootstrap/README.md
- https://github.com/rust-lang/rust/blob/master/src/bootstrap/lib.rs

It would be nice to consolidate these some; e.g. `lib.rs` can probably just point to the README, and lots of the information in the readme is duplicated in the dev-guide and can be deleted and replaced with a link.

cc @Mark-Simulacrum - do you have strong opinions on this?

### PR DESCRIPTION
**Pull request by onur-ozkan**
With this diff, I tried to consolidate bootstrap documentations and remove the duplicated informations.

Coupled with https://github.com/rust-lang/rustc-dev-guide/pull/1563

Resolves #90686

Signed-off-by: ozkanonur <work@onurozkan.dev>

### CODE REVIEW

**Comment by albertlarsan68** [comment]
I think those changes should wait for the `rustc-dev-guide` to be updated, to be sure that there is not a moment where the documentation does not exist anywhere.
@rustbot blocked

File: src/bootstrap/README.md [hunk 1]
```diff
@@ -322,6 +206,11 @@ building rustc from source don't need an update to `VERSION`.
 
 If you have any questions, feel free to reach out on the `#t-infra` channel in
 the [Rust Zulip server][rust-zulip] or ask on internals.rust-lang.org. When
-you encounter bugs, please file issues on the rust-lang/rust issue tracker.
+you encounter bugs, please file issues on [Rust issue tracker][rust-issue-tracker].
```
Comment (albertlarsan68) [inline src/bootstrap/README.md:209]:
```suggestion
you encounter bugs, please file issues on the [Rust issue tracker][rust-issue-tracker].
```

**Comment by ozkanonur** [comment]
https://github.com/rust-lang/rustc-dev-guide/pull/1563#event-8402723234 is merged, so this is no longer blocked.

File: src/bootstrap/README.md [hunk 2]
```diff
@@ -115,24 +52,6 @@ compiler. What actually happens when you invoke rustbuild is:
...
-compilation for the stage0 -> stage1 compilation -- this is because
-the stage1 compiler is changing, and we don't try to cache and reuse
-incremental artifacts across different versions of the compiler.
-
-You can always drop the `--incremental` to build as normal (but you
-will still be using the local nightly as your bootstrap).
-
 ## Directory Layout
 
 This build system houses all output under the `build` directory, which looks
```
Comment (jyn514) [inline src/bootstrap/README.md:57]: This whole section looks partially duplicated in https://rustc-dev-guide.rust-lang.org/building/bootstrapping.html#directories-and-artifacts-generated-by-bootstrap. I think we should keep it here and remove the section in the dev guide.

cc @bjorn3 , I know you've said that section has been helpful to you in the past. Maybe we can link to here from the guide.

**Review by jyn514 — COMMENTED** [review]
This looks like a good start, thanks. I left a few suggestions below, and I also think we should link to this README from the dev guide.

File: src/bootstrap/README.md (hunk 2 above)
Comment (bjorn3) [reply src/bootstrap/README.md:57]: I don't recall having said that this section was useful in the past. I do think it makes sense to deduplicate, but I'm not sure which copy should remain and where.

File: src/bootstrap/README.md [hunk 3]
```diff
@@ -242,41 +161,6 @@ build/
...
-## Build tools
-
-We've actually got quite a few tools that we use in the compiler's build system
-and for testing. To organize these, each tool is a project in `src/tools` with a
-corresponding `Cargo.toml`. All tools are compiled with Cargo (currently having
-independent `Cargo.lock` files) and do not currently explicitly depend on the
-compiler or standard library. Compiling each tool is sequenced after the
-appropriate libstd/libtest/librustc compile above.
-
 ## Extending rustbuild
```
[the remark this answers is not in the document]
Comment (ozkanonur) [reply src/bootstrap/README.md:164]: Do you mean [this(Contributing to bootstrap)](https://rustc-dev-guide.rust-lang.org/building/bootstrapping.html#contributing-to-bootstrap) section?

[the remark this answers is not in the document]
Comment (jyn514) [reply src/bootstrap/README.md:164]: No, this section: https://rustc-dev-guide.rust-lang.org/building/bootstrapping.html#contributing-to-bootstrap

[the remark this answers is not in the document]
Comment (ozkanonur) [reply src/bootstrap/README.md:155]: I updated it, however I am not so sure if we should cut that part from dev guide to here. Isn't good to have such information on the dev guide? It looks pretty user-face information.

File: src/bootstrap/README.md [hunk 4]
```diff
@@ -4,105 +4,32 @@ This is an in-progress README which is targeted at helping to explain how Rust
 is bootstrapped and in general, some of the technical details of the build
 system.
 
-## Using rustbuild
+Note that this README only covers internal informations, not about
+how to use the system. Please check [bootstrapping dev guide][bootstrapping-dev-guide]
+for further information.
```
Comment (jyn514) [inline src/bootstrap/README.md:9]:
```suggestion
Note that this README only covers internal information, not how to use the tool. Please check [bootstrapping dev guide][bootstrapping-dev-guide] for further information.
```

File: src/bootstrap/README.md [hunk 5]
```diff
@@ -236,63 +145,31 @@ build/
     # system will link (using hard links) output from stageN-{std,rustc} into
     # each of these directories.
     #
-    # In theory, there is no extra build output in these directories.
+    # In theory these are working rustc sysroot directories, meaning there is
+    # no extra build output in these directories.
```
Comment (jyn514) [inline src/bootstrap/README.md:149]: Not sure we need to mention this in the docs, but I'm hoping at some point to distinguish these "public" sysroots from the one's we're using  for bootstrap internally: https://rust-lang.zulipchat.com/#narrow/stream/326414-t-infra.2Fbootstrap/topic/Performing.20.60.2E.2Fx.2Epy.20dist.60.20results.20in.20a.20broken.20compiler.20at.20.2E.2E.2E

File: src/bootstrap/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (jyn514) [reply src/bootstrap/README.md:155]: I don't think so? These are instructions for what to do if you're adding a new option to bootstrap, which the vast majority of people shouldn't need to do.

**Comment by jyn514** [comment]
Thank you!!

@bors r+ rollup

**Pull request merged**